src/lib/server/result_serializer.cpp

```diff
@@ -0,0 +1,122 @@
+#include "result_serializer.hpp"
+#include "lossy_cast.hpp"
+
+namespace opossum {
+
+template <typename SocketType>
+void ResultSerializer::send_table_description(
+    std::shared_ptr<const Table> table,
+    const std::shared_ptr<PostgresProtocolHandler<SocketType>>& postgres_protocol_handler) {
+  // Calculate sum of length of all column names
+  uint32_t column_name_length_sum = 0;
+  for (auto& column_name : table->column_names()) {
+    column_name_length_sum += column_name.size();
+  }
+
+  postgres_protocol_handler->send_row_description_header(column_name_length_sum,
+                                                         static_cast<uint16_t>(table->column_count()));
+
+  for (ColumnID column_id{0u}; column_id < table->column_count(); ++column_id) {
+    uint32_t object_id = 0;
+    int16_t type_width = 0;
+
+    // Documentation of the PostgreSQL object_ids can be found at:
+    // https://crate.io/docs/crate/reference/en/latest/interfaces/postgres.html
+    switch (table->column_data_type(column_id)) {
+      case DataType::Int:
+        object_id = 23;
+        type_width = 4;
+        break;
+      case DataType::Long:
+        object_id = 20;
+        type_width = 8;
+        break;
+      case DataType::Float:
+        object_id = 700;
+        type_width = 4;
+        break;
+      case DataType::Double:
+        object_id = 701;
+        type_width = 8;
+        break;
+      case DataType::String:
+        object_id = 25;
+        type_width = -1;
```

Comment:
Where did you find these values?

Reply:
Run the query `select oid, typlen, typname from pg_catalog.pg_type order by oid;` in PostgreSQL. The types and oids are not well documented.